### ISSUE TITLE
BUG: Library does not use keep alive

### ISSUE THREAD

**Issue by cross311** [Azure/azure-storage-node#358]
Azure-storage-node uses the `request` library in its request-wrapper, however it is not configured to used keep-alive by default.

If running this library in azure and doing heavy use of blob storage, socket limitations will be reached.

A very simple configuration that solves the issue is adding this globally:
```js
var http = require('http');
var https = require('https');
var HttpAgent = require('agentkeepalive');
var HttpsAgent = require('agentkeepalive').HttpsAgent;

// https://github.com/node-modules/agentkeepalive#new-agentoptions
const agentOptions = {
  keepAlive: true,
  maxSockets: 16,
  maxFreeSockets: 16,
  keepAliveMsecs: 1000,
  freeSocketKeepAliveTimeout: 1000 * 60 * 5,
};

http.globalAgent = new HttpAgent(agentOptions);

https.globalAgent = new HttpsAgent(agentOptions);
```

This library can also set the agent on each request internally in the request-wrapper.

It would be nice if by default this library added some form of keep alive and/or made it configurable.
Given the code above if the app just runs before making requests then all sockets will be kept alive.

Not using keep alive will led to econnresets and connection timeouts in the azure network.

**Comment by jiacfan** [comment]
Hi, @cross311 

Thanks for mentioning this issue, we are planning a minor release for switching to use keep-alive, and have fulfilled some pre-validation. Will return to this thread, when the optimization is done!

Appreciate any other suggestions!

Thanks,
Jiachen

**Comment by jiacfan** [comment]
Hi, @cross311 

FYI. Keep-alive has been enabled by default in latest release.

Thanks,
Jiachen

**Comment by cross311** [comment]
fixed by: #368 and released in #383
